cou/utils/nova_compute.py

```diff
@@ -60,3 +64,25 @@ async def get_instance_count(unit: str, model: COUModel) -> int:
         f"No valid instance count value found in the result of {action_name} action "
         f"running on '{unit}': {action.results}"
     )
+
+
+async def get_instance_count_to_upgrade(unit: ApplicationUnit, model: COUModel) -> None:
+    """Get the instance count of a unit and enable the scheduler if it cannot upgrade.
+
+    :param unit: Unit to check if there are VMs running.
+    :type unit: ApplicationUnit
+    :param model: COUModel
+    :type model: COUModel
+    :raises HaltUpgradeExecution: When a unit has VMs running.
+    """
+    unit_instance_count = await get_instance_count(unit.name, model)
+    if unit_instance_count != 0:
+        await model.run_action(unit_name=unit.name, action_name="enable", raise_on_failure=True)
```

Comment:
I thought we agreed that this needs to be caught and log.error message should be provided in that situation.